### PR TITLE
KT-12195 Quickfix @JvmStatic on main() method in an object

### PR DESCRIPTION
**Pull request by t-kameyama**
https://youtrack.jetbrains.com/issue/KT-12195

### CODE REVIEW

File: idea/resources/intentionDescriptions/AddJvmStaticIntention/after.kt.template [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<spot>@JvmStatic</spot>
```
Comment (yole) [inline idea/resources/intentionDescriptions/AddJvmStaticIntention/after.kt.template]: The template shows a top-level method, not a method inside an `object`. Please update it to show what the intention action actually does.

File: idea/src/org/jetbrains/kotlin/idea/intentions/AddJvmStaticIntention.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+class AddJvmStaticIntention : SelfTargetingIntention<KtNamedFunction>(
+    KtNamedFunction::class.java,
+    "Add '@JvmStatic' annotation"
+), LowPriorityAction {
+
+    private val annotationFqName = FqName("kotlin.jvm.JvmStatic")
+
+    override fun isApplicableTo(element: KtNamedFunction, caretOffset: Int): Boolean {
+        if (element.findAnnotation(annotationFqName) != null) return false
+        if (!UnusedSymbolInspection.isEntryPoint(element)) return false
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/intentions/AddJvmStaticIntention.kt]: See `KotlinRunLineMarkerContributor` for the correct way to check that a function is `main`.

File: idea/src/org/jetbrains/kotlin/idea/intentions/AddJvmStaticIntention.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        element.parent.parent.run {
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/intentions/AddJvmStaticIntention.kt]: Please use `KtDeclaration.containingClassOrObject`.

**Comment by t-kameyama** [comment]
I fixed. Please review.

**Comment by yole** [comment]
Thanks a lot for your contribution!

**Pull request merged**